gobblin-metrics/src/main/avro/MetricReport.avro

```diff
@@ -0,0 +1,37 @@
+// Avro Schema for MetricReport objects used by gobblin-metrics
+// Compile with:
+// java -jar /path/to/avro-tools-1.7.7.jar compile -string schema gobblin-metrics/src/main/avro/MetricReport.avro gobblin-metrics/src/main/gen-java
+{
+  "type": "record",
+  "name": "MetricReport",
+  "namespace": "gobblin.metrics",
+  "fields" : [
+    {
+      "name": "tags",
+      "type": {
+        "type": "map",
+        "values": "string"
+      },
+      "doc": "Tags associated with the metrics."
+    },
+    {
+      "name": "timestamp",
+      "type": "long",
+      "doc": "Tags associated with the metrics."
+    },
+    {
+      "name": "metrics",
+      "type": {
+      "type": "array", "items": {
+        "type": "record",
+        "name": "Metric",
+        "fields": [
+          {"name": "name", "type": "string", "doc": "Metric name."},
+          {"name": "value", "type": "double", "doc": "Metric value."}
```

Comment:
We did not think it made sense to support any non-numeric type. With respect to numeric types, double supports full precision for most types (tiny loss of precision for long). The use case for metrics is mostly tracking and alerting, which don't need the 1e-18 loss of precision that can come from only using double.

Using exclusively double simplifies code, makes it easier to read / understand / use.